draft-ietf-teep-protocol.md

```diff
@@ -1501,21 +1503,31 @@ the selected TEEP cipher suite MUST be used in both directions.
 ## EATs and SUIT Reports {#eat-suit-ciphersuite}
 
 TEEP uses COSE for confidentiality of EATs and SUIT Reports sent by a TEEP Agent.
-EATs and SUIT Reports sent by a TEEP Agent MUST support the cipher suite
-listed below, and MAY support other algorithms.
+A TAM MUST support both of the cipher suites defined below, which are defined to
+be consistent with profiles listed in {{I-D.moran-suit-mti}}.  A TEEP Agent MUST
+support at least one of the two but can choose which one.  For example, a TEEP
+Agent might choose a given cipher suite if it has hardware support for it.
+A TAM or TEEP Agent MAY also support other algorithms in the COSE Algorithms
+registry in addition to the mandatory ones listed below.  It MAY also support use
+with COSE_Encrypt or other COSE types in additional cipher suites.
 
 ~~~~
-$eat-suit-cipher-suite /= eat-suit-cipher-suite-encrypt0-aesccm
+$eat-suit-cipher-suite /= eat-suit-cipher-suite-encrypt0-hpke-es256
 
-eat-suit-cipher-suite-encrypt0-aesccm = [ teep-operation-encrypt0-aesccm ]
+eat-suit-cipher-suite-encrypt0-hpke-es256 = [ teep-operation-encrypt0-hpke-es256 ]
+eat-suit-cipher-suite-encrypt0-hpke-eddsa = [ teep-operation-encrypt0-hpke-eddsa ]
```

Comment:
The `teep-operation-encrypt0-hpke-es256` and `teep-operation-encrypt0-hpke-eddsa` are not defined in the cddl file for the teep protocol, and they will have cddl syntax error.
So my thought in this link was
https://github.com/ietf-teep/teep-protocol/issues/286#issuecomment-1461324516
having the `eat-suit-cipher-suite-encrypt0-hpke-es256` and `eat-suit-cipher-suite-encrypt0-hpke-eddsa` being defined in the cddl file of the I-D.moran-suit-mti would be nice.
Then we do not have to add the identical definitions on both suit and teep.

Reply:
If there is cddl file of I-D.moran-suit-mti then we could just reference them by downloading the cddl file with wget or curl and concatenate them with teep's cddl file as what we are doing in the Makefile for others.

Reply:
I'd like to understand how you expect `eat-suit-cipher-suite-encrypt0-hpke-es256` to be defined. It is the aggregate of a group of COSE algorithm identifiers, so I am struggling to see how that would be defined in CDDL.

Even COSE's CDDL makes this hard. The relevant part of COSE's CDDL is:

```
Generic_Headers = (
    ? 1 => int / tstr,  ; algorithm identifier
    ? 2 => [+label],    ; criticality
    ? 3 => tstr / int,  ; content type
    ? 4 => bstr,        ; key identifier
    ? 5 => bstr,        ; IV
    ? 6 => bstr,        ; Partial IV
    ? 7 => COSE_Signature / [+COSE_Signature] ; Counter signature
)
```

What are the COSE semantics for "When defined within `#6.98(`, a header with key `1` must have value `-7`"?

Reply:
> The `teep-operation-encrypt0-hpke-es256` and `teep-operation-encrypt0-hpke-eddsa` are not defined in the cddl file for the teep protocol, and they will have cddl syntax error. 

Fixed, thanks.

> So my thought in this link was [#286 (comment)](https://github.com/ietf-teep/teep-protocol/issues/286#issuecomment-1461324516) having the `eat-suit-cipher-suite-encrypt0-hpke-es256` and `eat-suit-cipher-suite-encrypt0-hpke-eddsa` being defined in the cddl file of the I-D.moran-suit-mti would be nice. Then we do not have to add the identical definitions on both suit and teep.

Can do that in future if I-D.moran-suit-mti does so.

Reply:
I agree with Akira that draft-moran-suit-mti should define the ciphersuite identifiers because otherwise we have to explain and reference the algorithms again in this document.

Reply:
I am fine with the PR, just this portion of cddl syntax from the suit-mti generates the error which is beyond my knowledge of how to solve.
```
; Note: This is not the same definition as is used in COSE.
; It restricts a COSE header definition further without
; repeating the COSE definition. It should be merged
; with COSE by using the CDDL .and operator.
SUIT_COSE_Profile_Headers<id> = (
    protected : bstr .cbor {
        1 => id,
        int => any
    }
    unprotected : {
        int => any
    }
)
```
The error message from cddl tool.
```
*** Look for syntax problems around the %%% markers:
t
; repeating the COSE definition. It should be merged
; with COSE by using the CDDL .and operator.
%%%SUIT_COSE_Profile_Headers<id> = (
    protected : bstr .cbor %%%{
        1 => id,
        int => any
    }
    unprotected : {
        int => any
    }
)
```
How about tackling this at the IETF 116 hackathon?